docs/e2e-testing.md

```diff
@@ -0,0 +1,208 @@
+
+### Guidelines
+
+✅ Recommended readable test names
+
+```javascript
+- adds Bob to the address book
+- send 1 TST to Bob
+```
+
+⚠️ Test names remain approachable, recommend to avoid using the meaningless prefix 'should'
+
+```javascript
+- should add Bob to the address book
+- should send 1 TST to Bob
+```
+
+❌ Test name should be completely avoided: The use of a `should` prefix and the word `and` can decrease the readability of the test,making it harder to understand what the test is doing as well as diagnose and fix issues.
+
+```javascript
+- should add Bob to the address book and send 1 TST to Bob
+```
+
+✅ Recommended readable test names
+
+```javascript
+- removes an account imported with a private key
+- impossible to remove an account generated from the SRP imported in onboarding
+
+- the UI environment is locked down
+- the background environment is locked
+```
+
+❌ Test name should be completely avoided
+
+```javascript
+// in file: test/e2e/tests/add-account.spec.js
+- should be possible to remove an account imported with a private key, but should not be possible to remove an account generated from the SRP imported in onboarding
+
+// in file: test/e2e/tests/lockdown.spec.js
+- the UI and background environments are locked down
+```
+
+## Organization of test files
+
+It's essential to keep test files organised. As test suites get bigger, a well-structured organisation makes it easier to search for tests as well as identify logical groups of tests that may be impacted by changes to an area of the application. When tests are organised based on related features or functionality it becomes easier to identify common helper functions that can be shared across the tests, reducing duplication.
+
+Organise tests into folders based on scenarios and features. This means that each type of scenario has its own folder, and each feature team owns one or more folders.
+
+- This approach assigns ownership of E2E testing at the feature team level. Each feature team, being aware of the features they own, finds it easier to understand the existing tests and identify what's missing.
+- The future goal is for specific feature teams to manage the necessary test cases for their own features, instead of the core teams tracking each one.
+- Using this organizational strategy eliminates the need to decide at a low level where to place a test, as it is straightforward based on the feature or scenario.
+- This approach aligns with the testing strategy for the mobile app. The mobile team prefers to have feature team tests under the same folder. This reduces friction when switching context.
+
+Example for organization of test files by features and scenarios:
+
+```javascript
+// current test path:
+test/e2e/tests/nft/import-erc1155.spec.js
+// recommended test path: (consolidate all import tests for different tokens into a single repository)
+test/e2e/tests/tokens/import/import-erc1155.spec.js
+
+// current test path:
+test/e2e/tests/clear-activity.spec.js
```

Comment:
```suggestion
metamask-extension/test/e2e/tests/clear-activity.spec.js
```

Reply:
As discussed in our 1<>1, we opted to omit the root folder in the relative path and specify the path of the e2e tests. I have updated this.